### PR TITLE
Open editor updates

### PR DESCRIPTION
**Pull request by gleachkr**
Here's the PR. I removed one of the builtins (`urxvtc`), since always exits immediately (just telling `urxvtd` to open a terminal) and won't work properly with callbacks. I figure the `urxvt` builtin should be sufficient.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -10,6 +10,7 @@
 ### Changed
 
  - `editor.lua` now uses substitution strings, rather than `global` to determine which editor to open.
+ - `open_editor.lua` now uses `editor.lua` rather than always using `xdg-open`
```
Comment (aidanholm) [inline CHANGELOG.md]: Should end in a full stop.

File: lib/editor.lua [hunk 2]
```diff
@@ -59,17 +57,17 @@ _M.editor_cmd = _M.builtin.default
 --- Edit a file in a terminal editor in a new window.
 --
 -- * Can't yet handle files with special characters in their name.
--- * Can't determine when text editor is closed.
 --
 -- @tparam string file The path of the file to edit.
 -- @tparam[opt] number line The line number at which to begin editing.
-_M.edit = function (file, line)
+-- @tparam[opt] function callback A callback that fires when the process spawned by the editor command exits.
+_M.edit = function (file, line,callback)
```
Comment (aidanholm) [inline lib/editor.lua]: Space between , and callback

File: lib/editor.lua [hunk 3]
```diff
@@ -59,17 +57,17 @@ _M.editor_cmd = _M.builtin.default
...
     local subs = {
         file = file,
         line = line or 1,
     }
     local cmd = string.gsub(_M.editor_cmd, "{(%w+)}", subs)
-    luakit.spawn(cmd)
+    luakit.spawn(cmd,callback)
```
Comment (aidanholm) [inline lib/editor.lua]: Ditto

File: lib/editor.lua [hunk 4]
```diff
@@ -59,17 +57,17 @@ _M.editor_cmd = _M.builtin.default
 --- Edit a file in a terminal editor in a new window.
 --
 -- * Can't yet handle files with special characters in their name.
--- * Can't determine when text editor is closed.
 --
 -- @tparam string file The path of the file to edit.
 -- @tparam[opt] number line The line number at which to begin editing.
-_M.edit = function (file, line)
+-- @tparam[opt] function callback A callback that fires when the process spawned by the editor command exits.
```
Comment (aidanholm) [inline lib/editor.lua]: Add "Arguments passed to the callback are the same as those passed to the `luakit.spawn` callback."

**Comment by gleachkr** [comment]
OK, fixed.

File: lib/editor.lua [hunk 5]
```diff
@@ -59,17 +57,17 @@ _M.editor_cmd = _M.builtin.default
 --- Edit a file in a terminal editor in a new window.
 --
 -- * Can't yet handle files with special characters in their name.
--- * Can't determine when text editor is closed.
 --
 -- @tparam string file The path of the file to edit.
 -- @tparam[opt] number line The line number at which to begin editing.
-_M.edit = function (file, line)
+-- @tparam[opt] function callback A callback that fires when the process spawned by the editor command exits. Arguments passed to the callback are the same as those passed to the `luakit.spawn` callback
```
Comment (aidanholm) [inline lib/editor.lua]:
```
FAIL style/test_luacheck / test_luacheck
  tests/style/test_luacheck.lua:102: Luacheck messages:
    lib/editor.lua:63: (1:202):           line is too long (202 > 120)
```

Comment (aidanholm) [inline lib/editor.lua]: Oh and it's missing a full stop at the end

Comment (gleachkr) [inline lib/editor.lua]: Drat. Sorry about that.

**Comment by gleachkr** [comment]
OK, tests now pass, and full stop has been added.

**Comment by aidanholm** [comment]
Looks good, thanks!

**Pull request merged**